### ISSUE TITLE
runMalt.sh tweak

### ISSUE THREAD

**Issue by Liontooth** [sammthomson/semafor#12]
Should bin/runMalt.sh have a path for java on line 71?
~~~
time ${JAVA_HOME_BIN}/java -Xmx2g -jar maltparser-1.7.2.jar ...
~~~
BTW we're experimenting with -XX:+UseNUMA -- would you expect a speed-up? We're not seeing any improvements in median speed, but the occasional slower processing runs vanish. I don't know if there's a simple way to test for a NUMA-capable machine.

Cheers,
David

**Comment by sammthomson** [comment]
On Wed, Mar 11, 2015 at 1:56 AM, David Liontooth <notifications@github.com>
wrote:

> Should bin/runMalt.sh have a path for java on line 71?
>
> time ${JAVA_HOME_BIN}/java -Xmx2g -jar maltparser-1.7.2.jar
>
Yes, that's better.

> BTW we're getting improved results if we use -XX:UseNUMA -- I don't know
> if there's a cheap check for NUMA-capable systems.
>
Oh, interesting! Improved results meaning faster runtime, right? I don't
know if there's a general way to check either. If not, we could make it a
user-controlled setting in `config.sh`, or just mention it in the readme.

**Comment by Liontooth** [comment]
It's a moderately faster runtime -- what we're seeing is a regularization of runtimes. Without NUMA we get spikes of really slow runs that take twice as long. With NUMA, these disappear, while the mode stays the same. This suggests NUMA improves memory management.

Note the correct syntax is -XX:+UseNUMA.

**Comment by sammthomson** [comment]
Fixed in 102e0c2d09618bbaf7ab3f924e0a8736d23e32dd and 20a58b732e181d3087da18aaf902dac9a2cdd36a (I added a note in the README). Thanks!
